app/services/data_sync/elasticsearch/user.rb

```diff
@@ -0,0 +1,44 @@
+module DataSync
+  module Elasticsearch
+    class User
+      RELATED_DOCS = %i[
+        articles
+        reactions
+        podcast_episodes
+        chat_channel_memberships
+        comments
+      ].freeze
+
+      attr_accessor :user, :updated_fields
+
+      delegate :articles, :chat_channel_memberships, :comments, to: :@user
+
+      def initialize(user, updated_fields)
+        @user = user
+        @updated_fields = updated_fields.deep_symbolize_keys
+      end
+
+      def sync_documents
+        return unless sync_needed?
+
+        RELATED_DOCS.each do |relation_name|
+          send(relation_name).find_each(&:index_to_elasticsearch)
+        end
+      end
+
+      private
+
+      def sync_needed?
+        updated_fields.key?(:username) || updated_fields.key?(:name)
```

Comment:
Why do we only care about `username` or `name`?

Reply:
Actually, you make a very good point. Right now we only care about those bc those are the only ones we search. However, we do have profile image URL and pro(though not used) in all of these docs. We should probably just check all of those right from the start.

Reply:
We could probably move those keys out to a constant and do something like:

```ruby
# Probably a better constant name
USER_FIELDS = %i[
  username
  name
  profile_image_url
]

def sync_needed?
  USER_FIELDS.any? { |key| updated_fields.key?(key) }
end

Reply:
Thanks 😄  went with `SHARED_USER_FIELDS`